glib-macros/src/genum_derive.rs

```diff
@@ -165,7 +165,12 @@ pub fn impl_genum(input: &syn::DeriveInput) -> TokenStream {
 
                 let name = std::ffi::CString::new(#gtype_name).expect("CString::new failed");
                 unsafe {
-                    let type_ = #crate_ident::gobject_sys::g_enum_register_static(name.as_ptr(), VALUES.as_ptr());
+                    // Lookup the type ID or return 0 if it has not yet been registered under the specific name
+                    let mut type_ = #crate_ident::gobject_sys::g_type_from_name(name.as_pr());
+                    if type_ == 0 {
+                        // Register the type ONLY if not done before
```

Comment:
This assumes that it's exactly the same enum, which is not guaranteed. Might be different versions or other conflicts.

The only safe solution here is to panic.

In your case I assume this happens because the enum registration code is statically linked multiple times into different pieces of code, so it gets registered multiple times. You must only have it exactly once by putting it into a shared library that is linked to all users of it.

Reply:
Thank you for your quick response. I'll try to determine where this static linking happens.

Reply:
If you have this in a crate that is a dependency of multiple GStreamer plugins plus maybe an application, then this would happen. Or if you link multiple different versions of the code with the enum into the same final result.